### PR TITLE
Cli Usability: made changes in the output format of tree command

### PR DESCRIPTION
**Pull request by NagaHarshita**
This pr makes the following changes in the tree command
* Updated the output format as mentioned in Option 2 of the discussion [#1850 ](https://github.com/GoogleContainerTools/kpt/issues/1850#issuecomment-833130968). Encloses the directory name inside the quotes if it is a directory. Adds Package as a prefix and encloses the package name inside quotes for a package. 
* tree command wouldn't accept stdin 
* Updated the unit tests according to the changes made 
* Updated the reference and book in the docs

### CODE REVIEW

File: thirdparty/cmdconfig/commands/cmdtree/tree.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ const (
 	// Resources packages.
 	TreeStructurePackage TreeStructure = "directory"
 
-	PkgPrefix = "PKG: "
+	PkgPrefix = "Package "
```
Comment (phanimarupaka) [inline thirdparty/cmdconfig/commands/cmdtree/tree.go:27]: This should be `PkgNameFormat = "Package %q:"` and describe it with comment what %q holds.

File: thirdparty/cmdconfig/commands/cmdtree/tree.go [hunk 2]
```diff
@@ -114,7 +114,7 @@ func branchName(root, dirRelPath string) string {
 	if !os.IsNotExist(err) {
 		// add Pkg: prefix indicating that it is a separate package as it has
 		// Kptfile
-		return PkgPrefix + name
+		return PkgPrefix + "\"" + name + "\":"
```
Comment (phanimarupaka) [inline thirdparty/cmdconfig/commands/cmdtree/tree.go:117]: `return fmt.Sprintf(PkgNameFormat, name)`

File: thirdparty/cmdconfig/commands/cmdtree/tree.go [hunk 3]
```diff
@@ -97,9 +97,9 @@ func (p TreeWriter) packageStructure(nodes []*yaml.RNode) error {
 				return err
 			}
 			out = strings.TrimPrefix(out, ".")
-			out = filepath.Base(d) + out
+			out = "\"" + filepath.Base(d) + "\":" + out
```
Comment (phanimarupaka) [inline thirdparty/cmdconfig/commands/cmdtree/tree.go:100]: lines 90-103 must be refactored.

```
_, err := os.Stat(filepath.Join(p.Root, kptfilev1alpha2.KptFileName))
	if !os.IsNotExist(err) {
		if p.Root == "." {
			// replace . with the current working dir name
			d, err := os.Getwd()
			if err != nil {
				return err
			}
			p.Root = filepath.Base(d)
		}
		tree.SetValue(fmt.Sprintf(PkgNameFormat, p.Root))
	}
	out := tree.String()
```

is cleaner

File: thirdparty/cmdconfig/commands/cmdtree/cmdtree_test.go [hunk 4]
```diff
@@ -305,12 +305,12 @@ openAPI:
 		return
 	}
 
-	if !assert.Equal(t, fmt.Sprintf(`PKG: %s
+	if !assert.Equal(t, fmt.Sprintf(`Package %s
```
Comment (phanimarupaka) [inline thirdparty/cmdconfig/commands/cmdtree/cmdtree_test.go:308]: `Package %q:`

**Review by phanimarupaka — CHANGES_REQUESTED** [review]

**Comment by phanimarupaka** [comment]
Please make sure you update the book https://kpt.dev/book/01-getting-started/02-quickstart with new output.

File: thirdparty/cmdconfig/commands/cmdtree/tree.go [hunk 5]
```diff
@@ -23,8 +23,8 @@ const (
 	// TreeStructurePackage configures TreeWriter to generate the tree structure off of the
 	// Resources packages.
 	TreeStructurePackage TreeStructure = "directory"
-
-	PkgPrefix = "PKG: "
+	// %q holds the package or directory name
```
Comment (phanimarupaka) [inline thirdparty/cmdconfig/commands/cmdtree/tree.go:26]: // %q holds the package name

File: thirdparty/cmdconfig/commands/cmdtree/cmdtree_test.go [hunk 6]
```diff
@@ -83,7 +83,7 @@ spec:
 		return
 	}
 
-	if !assert.Equal(t, `.
+	if !assert.Equal(t, `".":
```
Comment (phanimarupaka) [inline thirdparty/cmdconfig/commands/cmdtree/cmdtree_test.go:86]: This should print the directory name and not `.`

File: thirdparty/cmdconfig/commands/cmdtree/tree.go [hunk 7]
```diff
@@ -87,21 +86,30 @@ func (p TreeWriter) packageStructure(nodes []*yaml.RNode) error {
 		}
 	}
 
-	out := tree.String()
 	_, err := os.Stat(filepath.Join(p.Root, kptfilev1alpha2.KptFileName))
 	if !os.IsNotExist(err) {
 		if p.Root == "." {
-			// replace . with the current working dir name
+			// get the path to current working directory
```
Comment (phanimarupaka) [inline thirdparty/cmdconfig/commands/cmdtree/tree.go:92]: The code doesn't look the way I pasted. Please verify.

**Review by phanimarupaka — CHANGES_REQUESTED** [review]

File: site/book/01-getting-started/02-quickstart.md [hunk 8]
```diff
@@ -19,7 +19,7 @@ Next, let's quickly view the content of the package:
 
 ```shell
 $ kpt pkg tree
-Package nginx:
+Package "nginx":
```
Comment (frankfarzan) [inline site/book/01-getting-started/02-quickstart.md:22]: There are more references to tree in the docs. You can do a grep in the repo, or search through the UI: https://kpt.dev

Comment (phanimarupaka) [reply site/book/01-getting-started/02-quickstart.md:22]: The best way to fix this is to add verify guides mark up for the guides in book similar to reference docs. https://github.com/GoogleContainerTools/kpt/blob/next/site/reference/pkg/get/README.md#examples

You can do it as your next task.

Comment (NagaHarshita) [reply site/book/01-getting-started/02-quickstart.md:22]: I found 4 references of tree command in the book apart from command reference using the grep command. I manually checked the output and updated the docs accordingly.

File: site/book/02-concepts/01-packages.md [hunk 9]
```diff
@@ -13,13 +13,13 @@ Let's take a look at an example:
 ```shell
 $ kpt pkg get https://github.com/GoogleContainerTools/kpt.git/package-examples/wordpress@v0.3
 $ kpt pkg tree wordpress/
-PKG: wordpress
+Package "wordpress":
 ├── [Kptfile]  Kptfile wordpress
 ├── [service.yaml]  Service wordpress
-├── deployment
+├── "deployment":
```
Comment (frankfarzan) [inline site/book/02-concepts/01-packages.md:19]: why do we quote the directories and not other things like filenames? It seems inconsistent.

Related: @phanimarupaka did we ever consider alternative notations to presenting packages names? Not suggesting this per se but to 

```
[[wordpress]]
   [Kptfile]
```

Comment (phanimarupaka) [reply site/book/02-concepts/01-packages.md:19]: We can remove quotes for directory names if you want. 

I don't have strong opinion. I proposed few options here and felt this is better. If you have a preference please paste the expected output here and we can go ahead.

**Comment by phanimarupaka** [comment]
@NagaHarshita Here is the decided format.

```
$ kpt pkg tree wordpress/
Package "wordpress"
├── [Kptfile]  Kptfile wordpress
├── [service.yaml]  Service wordpress
├── deployment
│   ├── [deployment.yaml]  Deployment wordpress
│   └── [volume.yaml]  PersistentVolumeClaim wp-pv-claim
└── Package "mysql"
    ├── [Kptfile]  Kptfile mysql
    ├── [deployment.yaml]  PersistentVolumeClaim mysql-pv-claim
    ├── [deployment.yaml]  Deployment wordpress-mysql
    └── [deployment.yaml]  Service wordpress-mysql
```

Directory should not have quotes and `:`

Also update this README as well with new pattern https://github.com/GoogleContainerTools/kpt/tree/next/package-examples/wordpress

File: thirdparty/cmdconfig/commands/cmdtree/cmdtree_test.go [hunk 10]
```diff
@@ -83,12 +83,12 @@ spec:
 		return
 	}
 
-	if !assert.Equal(t, `.
+	if !assert.Equal(t, fmt.Sprintf(`%q
```
Comment (phanimarupaka) [inline thirdparty/cmdconfig/commands/cmdtree/cmdtree_test.go:86]: %q ? It should be %s correct if it is directory with no Kptfile

Comment (NagaHarshita) [reply thirdparty/cmdconfig/commands/cmdtree/cmdtree_test.go:86]: Yes, will change that.

**Review by phanimarupaka — APPROVED** [review]

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**